docs/okeps/okep-5193-user-defined-networks.md

```diff
@@ -0,0 +1,1468 @@
+Although this is an unlikely scenario, it is good to document.
+
+### Network Policy
+
+Network Policy will be fully supported for user-defined primary networks as it is today with the cluster default network.
+However, configuring network policies that allow traffic between namespaces that connect to different user-defined
+primary networks will have no effect. This traffic will not be allowed, as the networks have no connectivity to each other.
+These types of policies will not be invalidated by OVN-Kubernetes, but the configuration will have no effect. Namespaces
+that share the same user-defined primary network will still benefit from network policy that applies access control over
+a shared network. Additionally, policies that block/allow cluster egress or ingress traffic will still be enforced for
+any user-defined primary network.
+
+### API Details
+
+Network Attachment Definitions (NADs) are the current way to configure the network in OVN-Kubernetes today, and the
+method proposed in this enhancement. There are two major shortcomings of NAD:
+
+1. It has free-form configuration that depends on the CNI. There is no API validation of what a user enters, leading to
+   mistakes which are not caught at configuration time and may cause unexpected functional behavior at runtime.
+2. It requires cluster admin RBAC in order to create the NAD.
+
+In order to address these issues, a proper CRD may be implemented which indirectly creates the NAD for OVN-Kubernetes.
+This solution may consist of more than one CRD, namely an Admin based CRD and one that is namespace scoped for tenants.
+The reasoning behind this is we want tenants to be able to create their own user-defined network for their namespace,
+but we do not want them to be able to connect to another namespace’s network without permission. The Admin based version
+would give higher level access and allow an administrator to create a network that multiple namespaces could connect to.
+It may also expose more settings in the future for networks that would not be safe in the hands of a tenant, like
+deciding if a network is able to reach other services in other networks. With tenants having access to be able to create
+multiple networks, we need to consider potential attack vectors like a tenant trying to exhaust OVN-Kubernetes
+resources by creating too many secondary networks.
+
+Furthermore, by utilizing a CRD, the status of the network CR itself can be used to indicate whether it is configured
+by OVN-Kubernetes. For example, if a user creates a network CR and there is some problem (like pods already existed) then
+an error status can be reported to the CR, rather than relying on the user to check OVN-Kubernetes logs.
+
+#### API Overview
+
+Following the [CRDs for Managing Networks](#crds-for-managing-networks), two CRDs shall be introduced:
+- Namespace scoped CRD - represent user request for creating namespace scoped OVN network.
```

Comment:
can also link to https://ovn-kubernetes.io/api-reference/userdefinednetwork-api-spec/ saying - see this for final API version